salt/crypt.py

```diff
@@ -806,7 +806,7 @@ def minion_sign_in_payload(self):
             pubkey_path = os.path.join(self.opts['pki_dir'], self.mpub)
             pub = get_rsa_pub_key(pubkey_path)
             if HAS_M2:
-                payload['token'] = pub.public_encrypt(six.b(self.token), RSA.pkcs1_oaep_padding)
+                payload['token'] = pub.public_encrypt(self.token, RSA.pkcs1_oaep_padding)
```

Comment:
Does `public_encrypt` want a bytestring for the first argument? If so then we should be using `to_bytes` here. It turns out that `six.b()` is a liar and it doesn't produce a bytestring on Python 2:

```python
>>> six.b(u'foo')
u'foo'
```

Reply:
Yes, `public_encrypt` wants a bytestring but `self.token` is already a bytestring so we don't need to convert anything here.

Reply:
OK, as long as the same holds true for the rest of the instances of six.b, we should be fine.

We should not be using six.b, since it doesn't do what it says it does.

Reply:
Got it. Thanks @terminalmage !